toc/rfc/rfc-draft-role-change-process.md

```diff
@@ -0,0 +1,95 @@
+# Meta
+[meta]: #meta
+- Name: Role Change Process
+- Start Date: 2022-04-13
+- Author(s): @gerg
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (fill in with PR link after you submit it)
+
+
+## Summary
+
+Define the process for granting or revoking roles.
+
+## Problem
+
+There are currently
+[descriptions](https://github.com/cloudfoundry/community/blob/main/toc/ROLES.md)
+for the criteria necessary to become a Contributor or Approver. It is not
+currently defined how people apply for or are granted these roles. This makes it
+difficult for community members to know how to apply for a role promotion and
+difficult for Working Group leadership to manage Contributor and Approver
+candidates.
+
+In addition, there is no defined process for removing people from Contributor or
+Approver roles if they are no longer interested in participating at that level.
+
+## Proposal
+
+### Promotion to Contributor
```

Comment:
Overlaps to a good extend https://github.com/cloudfoundry/community/blob/main/toc/rfc/rfc-0002-github-members.md

Reply:
I view this as an extension of that RFC. I'm primarily interested in the process through which the promotion is reviewed and accepted.

That said, maybe there is more I can do to integrate the two RFCs.

Reply:
For instance, I think this RFC currently conflicts with this paragraph:


> As removing a member from the `cloudfoundry` GitHub org may have unintended consequences across the organization, the TOC is the body required to approve those removals.
Proposals to remove a member should also be submitted via PR, and the submitter should tag the PR with `toc` and should mention `@cloudfoundry/toc` to make the TOC aware of the request.
The TOC will then consult any working groups that may be affected by the removal of this member and use its usual decision process to approve or reject the removal.

Reply:
I don't remember the exact reason why removing a member from the cloudfoundry org might have unintended consequences. Can't really imaging for a contributor and I agree that TOC should not be required here.

Maybe this discussion is caused by using a low level technical document (cloudfoundry.yml) for maintaining a higher level concept (contributor role).  Could be solved by having a separate contributor document that gets (eventually automated) translated into cloudfoundry.yml similar to the WG charters. Allows e.g. to ensure that all approvers are members of the cloudfoundry org and makes it easier for new contributors to file a PR: cloudfoundry.yml has currently 7k lines and ~650 org members.